File: apps/api/app/schemas/ai_settings.py
from datetime import datetime
from enum import StrEnum
from uuid import UUID

from pydantic import BaseModel, ConfigDict, SecretStr


class AiProvider(StrEnum):
    GEMINI = "gemini"
    OPENAI = "openai"


class AiSettingsStatus(BaseModel):
    configured: bool
    provider: AiProvider | None = None
    masked_hint: str | None = None
    updated_at: datetime | None = None
    updated_by: UUID | None = None
    updated_by_name: str | None = None

    model_config = ConfigDict(from_attributes=True)


class AiSettingsUpdateRequest(BaseModel):
    provider: AiProvider
    api_key: SecretStr

    model_config = ConfigDict(extra="forbid")


class AiSettingsError(BaseModel):
    code: str
    message: str

File: apps/web/lib/api/ai-settings.ts
import { apiFetch } from "./client";

export type AiProvider = "gemini" | "openai";

export type AiSettingsStatus = {
  configured: boolean;
  provider: AiProvider | null;
  masked_hint: string | null;
  updated_at: string | null;
  updated_by: string | null;
  updated_by_name?: string | null;
};

export type AiSettingsUpdateInput = {
  provider: AiProvider;
  apiKey: string;
};

export async function getAiSettings(workspaceId: string) {
  return apiFetch<AiSettingsStatus>(`/workspaces/${workspaceId}/ai-settings`);
}

export async function putAiSettings(workspaceId: string, input: AiSettingsUpdateInput) {
  return apiFetch<AiSettingsStatus>(`/workspaces/${workspaceId}/ai-settings`, {
    method: "PUT",
    body: JSON.stringify({
      provider: input.provider,
      api_key: input.apiKey,
    }),
  });
}

export async function deleteAiSettings(workspaceId: string) {
  return apiFetch<AiSettingsStatus>(`/workspaces/${workspaceId}/ai-settings`, {
    method: "DELETE",
  });
}
